jazz_deployments/components/validation/common.js

```diff
@@ -0,0 +1,152 @@
+
+/**
+    Common Validation functions for Environments-Catalog
+    @module: common.js
+    @description: Defines common validate functions.
+    @author:
+    @version: 1.0
+**/
+
+const _ = require("lodash");
+
+var validateIsEmptyInputData = function (deployment_data, onComplete) {
+    if (_.isEmpty(deployment_data)) {
+        onComplete({
+            result: "inputError",
+            message: "Input payload cannot be empty"
+        }, null);
+    } else {
+        onComplete(null, {
+            result: "success",
+            input: deployment_data
+        });
+    }
+};
+
+var validateAllRequiredFields = function (deployment_data, required_fields, onComplete) {
+    var missing_required_fields = _.difference(_.values(required_fields), _.keys(deployment_data));
+    if (missing_required_fields.length > 0) {
+        var message = "Following field(s) are required - " + missing_required_fields.join(", ");
+        onComplete({
+            result: "inputError",
+            message: message
+        }, null);
+    } else {
+        onComplete(null, {
+            result: "success",
+            input: deployment_data
+        });
+    }
+};
+
+var validateUnAllowedFieldsInInput = function (deployment_data, fields_list, onComplete) {
+    var invalid_fields = _.difference(_.keys(deployment_data), _.values(fields_list));
+    if (invalid_fields.length > 0) {
+        var message = "Following fields are invalid :  " + invalid_fields.join(", ") + ". ";
+        onComplete({
+            result: "inputError",
+            message: message
+        }, null);
+    } else {
+        onComplete(null, {
+            result: "success",
+            input: deployment_data
+        });
+    }
+};
+
+var validateAllRequiredFieldsValue = function (deployment_data, required_fields, onComplete) {
+    var invalid_required_fields = [];
+    _.forEach(required_fields, function (value, key) {
+        if (_.isEmpty(deployment_data[value])) {
+            invalid_required_fields.push(value);
+        }
+    });
+
+    if (invalid_required_fields.length > 0) {
+        var message = "Following field(s) value cannot be empty - " + invalid_required_fields.join(", ");
+        onComplete({
+            result: "inputError",
+            message: message
+        }, null);
+    } else {
+        onComplete(null, {
+            result: "success",
+            input: deployment_data
+        });
+    }
+};
+
+var validateRemoveEmptyValues = function (deployment_data, onComplete) {
+    for (var field in deployment_data) {
+        if (!deployment_data[field]) {
+            delete deployment_data[field];
+        }
+    }
+    onComplete(null, {
+        result: "success",
+        input: deployment_data
+    });
+};
+
+var validateNotEditableFieldsInUpdate = function (deployment_data, fields_list, onComplete) {
+    var invalid_fields = _.intersection(_.keys(deployment_data), _.values(fields_list));
+    _.forEach(invalid_fields, function (value, key) {
```

Comment:
remove lodash

Reply:
i think you already know :) list.forEach and better using ;ist.map

Reply:
Using map